src/ruleset.ts

```diff
@@ -775,24 +772,152 @@ export default {
+        },
+      ],
+    },
+
+    /**
+     * @author: Jason Harmon <https://github.com/jharmn>
+     */
+    "owasp:api8:2023-define-error-responses-500": {
+      message: "Operation is missing {{property}}.",
+      description:
+        "OWASP API Security recommends defining schemas for all responses, even errors. The 500 describes what happens when a request fails with an internal server error, so its important to define this not just for documentation, but to empower contract testing to make sure the proper JSON structure is being returned instead of leaking implementation details in backtraces.",
+      severity: DiagnosticSeverity.Warning,
+      given: "$.paths..responses",
+      then: [
+        {
+          field: "500",
+          function: truthy,
+        },
+        {
+          field: "500.content",
+          function: truthy,
+        },
+      ],
+    },
+
+    /**
+     * API9:2023 Improper Inventory Management
+     * https://owasp.org/API-Security/editions/2023/en/0xa9-improper-inventory-management/
      *
-     * Attackers construct API calls that include SQL, NoSQL, LDAP, OS, or other commands that the API or the backend behind it blindly executes.
+     * How to prevent
+     * - 🟠 Servers, define which environment is the API running in (e.g. production, staging, test, development)
+     * - ✅ Require servers use x-internal true/false to explicitly explain what is public or internal for documentation tools
+     * - 🤷‍♂️ There is no retirement plan for each API version.
+     */
+
+    /**
+     * @author: Phil Sturgeon <https://github.com/philsturgeon>
+     */
+    "owasp:api9:2023-inventory-access": {
+      message:
+        "Declare intended audience of every server by defining servers{{property}} as true/false.",
+      description:
+        "Servers are required to use vendor extension x-internal set to true or false to explicitly explain the audience for the API, which will be picked up by most documentation tools.",
+      severity: DiagnosticSeverity.Error,
+      formats: [oas3],
+      given: "$.servers.*",
+      then: {
+        field: "x-internal",
+        function: defined,
+      },
+    },
+
+    /**
+     * @author: Phil Sturgeon <https://github.com/philsturgeon>
+     */
+    "owasp:api9:2023-inventory-environment": {
+      message:
+        "Declare intended environment in server descriptions using terms like local, staging, production.",
+      description:
+        "Make it clear which servers are expected to run as which environment to avoid unexpected problems, exposing test data to the public, or letting bad actors bypass security measures to get to production-like environments.",
+      severity: DiagnosticSeverity.Error,
+      formats: [oas3],
+      given: "$.servers.*",
+      then: {
+        field: "description",
+        function: pattern,
+        functionOptions: {
+          match:
+            "/(local|alpha|beta|test|testing|stag|staging|prod|production|next|preprod)/i",
```

Comment:
```suggestion
            "/(local|sandbox|alpha|beta|test|testing|stag|staging|prod|production|next|preprod|preproduction)/i",
```
Adding `sandbox` and `preproduction` as other common environments I've encountered